File: ProjectTest/OneLang/Python/OneLang/One/Workspace.py
from OneLangStdLib import *
import OneLang.One.ErrorManager as errorMan
import OneLang.One.Ast.Types as types

class Workspace:
    def __init__(self):
        self.packages = {}
        self.error_manager = errorMan.ErrorManager()
    
    def add_package(self, pkg):
        self.packages[pkg.name] = pkg
    
    def get_package(self, name):
        pkg = self.packages.get(name)
        if pkg == None:
            raise Error(f'''Package was not found: "{name}"''')
        return pkg

File: ProjectTest/OneLang/CSharp/One/Workspace.cs
using One;
using One.Ast;
using System.Collections.Generic;

namespace One
{
    public class Workspace {
        public Dictionary<string, Package> packages;
        public ErrorManager errorManager;
        
        public Workspace()
        {
            this.packages = new Dictionary<string, Package> {};
            this.errorManager = new ErrorManager();
        }
        
        public void addPackage(Package pkg) {
            this.packages.set(pkg.name, pkg);
        }
        
        public Package getPackage(string name) {
            var pkg = this.packages.get(name);
            if (pkg == null)
                throw new Error($"Package was not found: \"{name}\"");
            return pkg;
        }
    }
}